crates/astria-conductor/src/data_availability/block_verifier.rs

```diff
@@ -296,6 +296,23 @@ mod test {
 
     use super::*;
 
+    /// Constructs a `[merkle::Tree]` from an iterator yielding byte slices.
+    ///
+    /// This hashes each item before pushing it into the Merkle Tree, which
+    /// effectively causes a double hashing. The leaf hash of an item `d_i`
+    /// is then `MTH(d_i) = SHA256(0x00 || SHA256(d_i))`.
+    fn merkle_tree_from_transactions<I, B>(iter: I) -> merkle::Tree
```

Comment:
no code changes here. Just moved this out of the sequencer-types crate as this was only used in conductor.